pkg/issuer/acme/dns/route53/route53.go

```diff
@@ -90,22 +99,78 @@ func NewDNSProvider(accessKeyID, secretAccessKey, hostedZoneID, region string, a
 		// https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html#specifying-credentials
 	} else {
 		klog.V(5).Infof("not using ambient credentials")
-		config.WithCredentials(credentials.NewStaticCredentials(accessKeyID, secretAccessKey, ""))
+		sessionOpts.Config.Credentials = credentials.NewStaticCredentials(d.AccessKeyID, d.SecretAccessKey, "")
 		// also disable 'ambient' region sources
 		sessionOpts.SharedConfigState = session.SharedConfigDisable
 	}
 
-	// If ambient credentials aren't permitted, always set the region, even if to
-	// empty string, to avoid it falling back on the environment.
-	if region != "" || !useAmbientCredentials {
-		config.WithRegion(region)
-	}
 	sess, err := session.NewSessionWithOptions(sessionOpts)
 	if err != nil {
 		return nil, fmt.Errorf("unable to create aws session: %s", err)
 	}
+
+	if d.Role != "" {
+		klog.V(5).Infof("assuming role: %s", d.Role)
+		stsSvc := d.StsProvider(sess)
+		result, err := stsSvc.AssumeRole(&sts.AssumeRoleInput{
+			DurationSeconds: aws.Int64(900),
```

Comment:
Not too sure how long we use that `GetSession()`.  A validation might take longer than 900 seconds and we might not cleanup properly. 

It also might be the case that we always get a new session on every reconcile (@munnerz to confirm)

Reply:
> It also might be the case that we always get a new session on every reconcile (@munnerz to confirm)

We do currently get a new session on each reconcile (although we don't *always* obtain a session on *every* reconcile, i.e. we only get one if we need to Present or CleanUp the record, so you *shouldn't* see 10s or hundreds of sessions being created for a single challenge)

Reply:
> A validation might take longer than 900 seconds and we might not cleanup properly.

Good point! 1h is the sdk default and should be enough.